### PR TITLE
#4998: Add backward support for hypot

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
Add backward support for hypot

### CODE REVIEW

**Comment by KalaivaniMCW** [comment]
CI test [Link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/7709699246)

**Review by muthutt — CHANGES_REQUESTED** [review]
please see comment; otherwise nice change - thanks @KalaivaniMCW

File: tt_eager/tt_dnn/op_library/backward/backward_ops.cpp [hunk 1]
```diff
@@ -501,6 +501,23 @@ std::vector<Tensor> relu_bw(const Tensor& grad, const Tensor& input, const Memor
     return operation::decorate_as_composite(__func__, _relu_bw)(grad, input, output_mem_config);
 }
 
+std::vector<Tensor> _hypot_bw(const Tensor& grad, const Tensor& input, const Tensor& other, const MemoryConfig& output_mem_config) {
+    std::vector<Tensor> grad_tensor;
+    Tensor pow_input = power(input, 2, output_mem_config);
```
[the remark this answers is not in the document]
Comment (KalaivaniMCW) [reply tt_eager/tt_dnn/op_library/backward/backward_ops.cpp]: Done

**Comment by KalaivaniMCW** [comment]
@muthutt Addressed the comment and made necessary changes

File: tt_eager/tt_dnn/op_library/backward/backward_ops.cpp [hunk 2]
```diff
@@ -501,6 +501,21 @@ std::vector<Tensor> relu_bw(const Tensor& grad, const Tensor& input, const Memor
     return operation::decorate_as_composite(__func__, _relu_bw)(grad, input, output_mem_config);
 }
 
+std::vector<Tensor> _hypot_bw(const Tensor& grad, const Tensor& input, const Tensor& other, const MemoryConfig& output_mem_config) {
+    std::vector<Tensor> grad_tensor;
+    Tensor hyp_result = hypot(input, other, output_mem_config);
+    Tensor result_recip = recip(hyp_result, output_mem_config);
+    Tensor grad_a = mul(grad, mul(input, result_recip, std::nullopt, output_mem_config), std::nullopt, output_mem_config);
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/backward/backward_ops.cpp:508]: You should scope/deallocate hyp_result after result_recip calculation

Comment (KalaivaniMCW) [reply tt_eager/tt_dnn/op_library/backward/backward_ops.cpp]: I have made changes to address your comment. Kindly review

**Review by muthutt — APPROVED** [review]
comments addressed - thanks @KalaivaniMCW

**Pull request merged**